metagpt/document_store/base_store.py

```diff
@@ -15,7 +15,7 @@ class BaseStore(ABC):
     """FIXME: consider add_index, set_index and think 颗粒度"""
 
     @abstractmethod
-    def search(self, query, *args, **kwargs):
+    def search(self, *args, **kwargs):
```

Comment:
no need to delete, we need an explicit definition of the search `query`. And the inherited search can be `search (self, query: List[float], collection_name: str, xxx)`.

Reply:
Although this will not report an error, it actually conflicts with the definition of the Base class. I think Base does not need to specify that there must be a query parameter, because the subclass must inherit and implement this search. As shown in the figure, pycharm will actually warn
![Uploading image.png…]()

Reply:
This is because the method signature of the abstract class search function changed by the search method I implemented, so this warning is reported